src/adapters/carbon-defi/products/strategies/carbonDeFiStrategiesAdapter.ts

```diff
@@ -0,0 +1,499 @@
+    } else if (!contractAddresses[this.chainId]) {
+      return []
+    }
+
+    return this.getCarbonMovements(
+      {
+        userAddress,
+        fromBlock,
+        toBlock,
+        tokenId,
+      },
+      'deposits',
+    )
+  }
+
+  /**
+   * Update me.
+   * Add logic to get tvl in a pool
+   *
+   */
+  async getTotalValueLocked(
+    _input: GetTotalValueLockedInput,
+  ): Promise<ProtocolTokenTvl[]> {
+    throw new NotImplementedError()
+  }
+
+  /**
+   * Update me.
+   * Add logic to calculate the underlying token rate of 1 protocol token
+   */
+  async getProtocolTokenToUnderlyingTokenRate(
+    _input: GetConversionRateInput,
+  ): Promise<ProtocolTokenUnderlyingRate> {
+    throw new NotImplementedError()
+  }
+
+  async getApy(_input: GetApyInput): Promise<ProtocolTokenApy> {
+    throw new NotImplementedError()
+  }
+
+  async getApr(_input: GetAprInput): Promise<ProtocolTokenApr> {
+    throw new NotImplementedError()
+  }
+
+  private processFilterStrategyUpdated(
+    strategyCreatedRawLog: StrategyCreatedEvent.Log,
+    strategyUpdatedLog: StrategyUpdatedEvent.Log[],
+    eventType: 'deposits' | 'withdrawals',
+  ): StrategyUpdatedEvent.Log[] {
+    const strategyUpdatedProcessed: StrategyUpdatedEvent.Log[] = []
+
+    for (let i = 0; i < strategyUpdatedLog.length; i++) {
+      const prevItem:
+        | StrategyCreatedEvent.OutputObject
+        | StrategyUpdatedEvent.OutputObject
+        | undefined =
+        i !== 0 ? strategyUpdatedLog[i - 1]?.args : strategyCreatedRawLog.args
+      const currentItem: StrategyUpdatedEvent.OutputObject | undefined =
+        strategyUpdatedLog[i]?.args
+
+      if (!prevItem || !currentItem) continue
+
+      const token0Diff = BigInt(
+        BigNumber(currentItem.order0.y.toString())
+          .minus(prevItem.order0.y.toString())
+          .toString(),
+      )
+      const token1Diff = BigInt(
+        BigNumber(currentItem.order1.y.toString())
+          .minus(prevItem.order1.y.toString())
+          .toString(),
+      )
+
+      const updatedItem = {
+        ...strategyUpdatedLog[i],
+        args: {
+          ...strategyUpdatedLog[i]!.args,
+          order0: { ...strategyUpdatedLog[i]!.args.order0, y: token0Diff },
+          order1: { ...strategyUpdatedLog[i]!.args.order1, y: token1Diff },
+        },
+      } as StrategyUpdatedEvent.Log
+
+      strategyUpdatedProcessed.push(updatedItem)
+    }
+
+    return strategyUpdatedProcessed.filter((item) => {
+      const currentItem: StrategyUpdatedEvent.OutputObject = item.args
+
+      return (
+        (eventType === 'deposits'
+          ? BigNumber(currentItem.order0.y.toString()).gt(0) ||
+            BigNumber(currentItem.order1.y.toString()).gt(0)
+          : BigNumber(currentItem.order0.y.toString()).lt(0) ||
+            BigNumber(currentItem.order1.y.toString()).lt(0)) &&
+        currentItem.reason === BigInt(StrategyUpdateReason.Edit)
```

Comment:
It looks like this is the only item of the enum we are using.

Could we instead of declaring an enum, create a constant such as:
`const StrategyUpdateReasonEdit = 0n`

That way no conversion is needed either.